apis/core/lib/domain/dataset/update.ts

```diff
@@ -0,0 +1,39 @@
+import { dcat, hydra, rdf, schema, _void } from '@tpluscode/rdf-ns-builders'
+import { GraphPointer } from 'clownface'
+import { NamedNode } from 'rdf-js'
+import { NotFoundError } from '../../errors'
+import { ResourceStore } from '../../ResourceStore'
+import { resourceStore } from '../resources'
+
+interface AddMetaDatatCommand {
+  dataset: GraphPointer<NamedNode>
+  resource: GraphPointer
+  store?: ResourceStore
+
+}
+
+export async function update({
+  dataset,
+  resource,
+  store = resourceStore(),
+}: AddMetaDatatCommand): Promise<GraphPointer> {
+  const datasetResource = await store.get(dataset.term)
+
+  if (!datasetResource) {
+    throw new NotFoundError(dataset)
+  }
+
+  datasetResource.deleteOut()
```

Comment:
This will not be enough because any deep blank nodes will stay in the graph. You'd need to be more thorough, to descend the blank nodes children. Something like this, without getting too sophisticated

```suggestion
  datasetResource.out().forEach(child => {
    if (child.term === 'BlankNode') {
      child.deleteOut()
    }
  })
  datasetResource.deleteOut()
```